vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/cached_discovery.go

```diff
@@ -231,8 +242,35 @@ func (d *CachedDiscoveryClient) ServerVersion() (*version.Info, error) {
 	return d.delegate.ServerVersion()
 }
 
+type cachedOpenAPIObjectEncoder struct{}
+
+func (e *cachedOpenAPIObjectEncoder) Encode(obj interface{}) ([]byte, error) {
+	return proto.Marshal(obj.(proto.Message))
+}
+
 func (d *CachedDiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
-	return d.delegate.OpenAPISchema()
+	filename := filepath.Join(d.cacheDirectory, "openapischema.json")
```

Comment:
You're forcing the encoding to proto, aren't you?

Reply:
> You're forcing the encoding to proto, aren't you?

Not exactly sure what you mean by this, but I am using `proto.Unmarshal` below to decode cached data.

To make things cleaner, I will rename the cache file to `openapischema.json.pb`